src/openjd/adaptor_runtime/_http/sockets.py

```diff
@@ -77,11 +84,19 @@ def get_socket_dir(self, namespace: str | None = None, *, create: bool = False)
                 not be raised if the user has configured a socket base directory.
         """
 
-        def create_dir(path: str) -> str:
-            if create:
+        def mkdir(path: str) -> str:
+            if create_dir:
                 os.makedirs(path, mode=0o700, exist_ok=True)
             return path
 
+        def gen_socket_path(dir: str, base_name: str):
+            i = 0
+            name = base_name
+            while os.path.exists(os.path.join(dir, name)):
+                i = i + 1
+                name = f"{base_name}_{i}"
```

Comment:
Would it make more sense to just use a UUID in order to avoid this iteration? I know it's not really a performance concern until there's thousands of sockets but if we can get rid of the loop altogether it seems worth it. Thoughts?

Reply:
Yeah that will work, but just have to be careful with the max length of socket names (108 bytes on Linux, 104 bytes on Mac). We would be restricting the max length of `base_name` to `MAX_SOCKET_NAME_LEN - len(UUID)`, but that's probably fine since it's plenty.

I'll push up a change to incorporate this

Reply:
I've decided to revert this change. A UUID has 32 characters without the hyphens, which will restrict the names of our sockets to much shorter lengths and I don't think this is worth the tradeoff for a small performance improvement.